src/main/java/org/apache/datasketches/tuple/CompactSketch.java

```diff
@@ -121,6 +121,11 @@
     }
   }
 
+  @Override
+  public CompactSketch<S> compact() {
+    return this;
+  }
```

Comment:
In order to do that we would have to add a method getSerializationBytes() or equivalent to the Summary Interface.  This is not a bad idea. Let me think about it.

Reply:
Great - just to add some detail - it is possible now to compute this by serialising the sketch and measuring the resulting byte size.  However, this is obviously wasteful and my question above is only predicated on whether the internal bytes are already available.

Reply:
Clearly doing toByteArray() would be very wasteful. The only knowledge we have of the Summary is the two current methods: we can copy it and serialize it to bytes.  When we are serializing the sketch we have to call toByteArray() anyway, so we grab its size at that point.  

But adding this method would give us that visibility and I think it could be useful.  However, we don't want to make the requirements on the Summary too restrictive as it may limit what the users want to do with it.  

Perhaps the best way to resolve this would be to put a [DISCUSS] issue on this topic on our dev@ list and see if anyone responds pro or con.

Reply:
This is the only item that is not resolved yet.

Reply:
I have looked through the additional changes, and yes, all the rest of the points raised were resolved.  Thank you for addressing them.  I have also read through the code related to the bug fixes mentioned and the additional commentary on clearing memory is valuable.

Concerning discussion thread here, I am interested in how the summary may be restricted.  Were you considering adding an additional method to the summary type to provide the size for the tuple?  
If it is more involved, I am happy to pick this up on [DISCUSS] or not let this block the PR.